packages/truffle-core/lib/testing/testrunner.js

```diff
@@ -30,21 +29,30 @@ function TestRunner(options) {
   this.first_snapshot = true;
   this.initial_snapshot = null;
   this.known_events = {};
-  this.web3 = new Web3();
+  this.web3 = options.legacy ? new LegacyWeb3() : new Web3();
   this.web3.setProvider(options.provider);
 
   // For each test
   this.currentTestStartBlock = null;
 
   this.BEFORE_TIMEOUT = 120000;
   this.TEST_TIMEOUT = 300000;
-};
+}
 
 TestRunner.prototype.initialize = function(callback) {
   var self = this;
 
+  if (self.web3.sha3) {
+    self.web3.utils = {};
+    self.web3.utils.sha3 = self.web3.sha3;
+  }
```

Comment:
(The stuff about `legacy-web3` is not me nitpicking, I swear. The comment I'm about to make here is nitpicking. This one you should feel free to roll your eyes at.)

This feels like a good candidate for moving to a function that has a semantic name, like `normalizeWeb3Interface(self.web3)` or `shimWeb3(self.web3)`. We could even put such a function inside my proposed `truffle-legacy-system` package :grin: